src/store/attribution.js

```diff
@@ -1,12 +1,16 @@
+import { defineStore } from 'pinia'
+
+import { ATTRIBUTION } from '~/constants/store-modules'
+import { COPY_ATTRIBUTION } from '~/constants/action-types'
+
 import GoogleAnalytics from '~/analytics/google-analytics'
 import { CopyAttribution } from '~/analytics/events'
-import { COPY_ATTRIBUTION } from '~/constants/action-types'
 
-export const createActions = (googleAnalytics) => ({
-  [COPY_ATTRIBUTION](_, params) {
-    const event = CopyAttribution(params.type, params.content)
-    googleAnalytics().sendEvent(event)
+export const useAttribution = defineStore(ATTRIBUTION, {
+  actions: {
+    [COPY_ATTRIBUTION](params) {
```

Comment:
Why would we continue using these constants with pinia? It makes the devex frustrating and defeats one of the best parts of Pinia, which is that the store is a well-typed object. Compare to:

```
export const useAttribution = defineStore('attribution', {
  actions: {
    copyAttribution(params) {
      // ...
    }
  }
}

const attribution = useAttribution()

attribution.copyAttribution()
```

This just works, and best, TypeScript is able to give you the correct types and hints when you do `attribution.`, the `.` will trigger auto-complete with the whole store's object.

If we could completely get rid of the store constants altogether that would be stellar in my opinion. With Pinia are there any reasons to keep them?

Reply:
This is true, I was going for the path of least change to identify if there was a way to migrate to Pinia while keeping the DX identical. Admittedly that was misguided and I've updated the PR accordingly.